spring-integration-core/src/main/java/org/springframework/integration/transaction/ExpressionEvaluatingTransactionSynchronizationProcessor.java

```diff
@@ -116,17 +115,23 @@ protected void onInit() {
 
 	@Override
 	public void processBeforeCommit(IntegrationResourceHolder holder) {
-		doProcess(holder, this.beforeCommitExpression, this.beforeCommitChannel, "beforeCommit");
+		if (this.beforeCommitChannel != null) {
```

Comment:
I don't this this is right solution.

We don't need to to build and send a message only when there is no particular channel.

Reply:
Sorry, I don't understand what you mean.

Reply:
Oh, I get it; sorry.

Reply:
Sorry, I meant: "I don't think".
My point was to call that `doProcess()` method don't build a reply message and don't send anything if there is no channel.
Not sure why would one build a message just to send it into a `NullChannel`...